src/test/java/ch/uzh/ifi/hase/soprafs23/controller/GroupControllerTest.java

```diff
@@ -275,6 +284,108 @@ public void testCreateGroupReturns404() throws Exception {
                 .andExpect(MockMvcResultMatchers.status().isNotFound());
     }
 
+    @Test
+    public void sendInvitation_returns404_whenGroupNotFound() throws Exception {
+        // given
+        Long groupId = 1L;
+        List<Long> guestIds = Arrays.asList(2L, 3L);
+        HttpServletRequest request = mock(HttpServletRequest.class);
+
+        Mockito.when(groupService.getGroupById(groupId)).thenThrow(new ResponseStatusException(HttpStatus.NOT_FOUND, String.format("Group with id %d not found.", groupId)));
+
+        String token = "valid-token";
+        Mockito.when(request.getHeader("X-Token")).thenReturn(token);
+
+        // when and then
+        mockMvc.perform(MockMvcRequestBuilders.post("/groups/" + groupId + "/invitations")
+                        .header("X-Token", token)
+                        .content(new ObjectMapper().writeValueAsString(guestIds))
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .accept(MediaType.APPLICATION_JSON))
+                .andExpect(MockMvcResultMatchers.status().isNotFound());
+    }
+
+
+    @Test
+    public void sendInvitation_returns401_whenUnauthorized() throws Exception {
+        // given
+        Long groupId = 1L;
+        List<Long> guestIds = Arrays.asList(2L, 3L);
+        HttpServletRequest request = mock(HttpServletRequest.class);
+
+        Group currentGroup = new Group();
+        currentGroup.setId(groupId);
+        currentGroup.setHostId(4L);
+
+        String token = "valid-token";
+        Long tokenId = 5L;
+
+        Mockito.when(groupService.getGroupById(groupId)).thenReturn(currentGroup);
+        Mockito.when(userService.getUseridByToken(request.getHeader("X-Token"))).thenReturn(tokenId);
+
+        // when and then
+        mockMvc.perform(MockMvcRequestBuilders.post("/groups/" + groupId + "/invitations")
+                        .header("X-Token", token)
+                        .content(new ObjectMapper().writeValueAsString(guestIds))
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .accept(MediaType.APPLICATION_JSON))
+                .andExpect(MockMvcResultMatchers.status().isUnauthorized());
+    }
+
```

Comment:
btw thats the sceleton of the test which is not working so far - even after hours of changing stuff lol - do it later